reactor-core/src/test/java/reactor/core/publisher/FluxBufferTimeoutTest.java

```diff
@@ -310,31 +312,54 @@ public void scanSubscriberCancelled() {
 
 	@Test
 	public void flushShouldNotRaceWithNext() {
-		Set<Integer> seen = new HashSet<>();
-		Consumer<List<Integer>> consumer = integers -> {
-			for (Integer i : integers) {
-				if (!seen.add(i)) {
-					throw new IllegalStateException("Duplicate! " + i);
+		for (int i = 0; i < 100; i++) {
```

Comment:
The changes in the test look like were introduced because the race testing utils are not trusted. At the same time, changing a test at the same time while introducing a change to the production code is suspicious - are these the final changes? Do we want to repeat the test 100 times?

Reply:
it is probability related thing. Just running race test once gives no guarantee of event to occur. JCStress are doing the same underneath (looping many times until a possible event is being triggered). Here, we do that manually. 

Ideally that should be moved to JCStress suit which does that in more sophisticated way + adds different VM configurations in addition which may create higher probability to a specific event occurrence.

Reply:
There should be a separate effort moving all such tests to JCStress. For now just wanna eliminate the bug